tools/New-AzureSampleResources.ps1

```diff
@@ -0,0 +1,169 @@
+    # This variable is used to load existing service principal details from another powershell script. For the existing details, the .ps1 file needs to contain
+    # a hashtable for the service principal application and certificate hashtable with the following keys accessible from this script:
+    # $azureAppSpInfo["ClientId"]
+    # $azureAppSpInfo["ClientSecret"]
+    # $azureAppSpInfo["ObjectId"]
+    # $azureCertSpInfo["ClientId"]
+    # $azureCertSpInfo["CertificateThumbprint"]
+    # $azureCertSpInfo["ObjectId"]
+    [Parameter(Mandatory = $false)]
+    [string] $ExistingServicePrincipalDetails
+)
+
+. "C:\CloudDeployment\BVTs\BVTSettingUtils.ps1"
+Import-Module $([io.path]::combine($PSScriptRoot, "Common.psm1"))
+Import-AzModules -AzVersion "0.10.0" 
+
+# Retrieve BVTSettings.xml settings.
+$adminName = Get-Setting -key ServiceAdminUpn
+$adminPassSecureString = ConvertTo-SecureString -String (Get-Setting -key ServiceAdminPassword) -AsPlainText -Force
+$deploymentId = Get-Setting -key DeploymentId
+$adfs = ([System.String](Get-Setting -key AadLoginUri)).EndsWith("/adfs")
+$azureTenantId = Get-Setting -key AADTenantID
+if ($adfs)
+{
+    $azureTenantId = "adfs"
+}
+$azureLocation = Get-Setting -Key AzureStackLocation
+$azureAdminArmEndpoint = Get-Setting -key ARMEndpoint
+
+# Log into Azure admin context
+Add-AzEnvironment -Name AzureSampleEnvironment -ARMEndpoint $azureAdminArmEndpoint
+$username = $adminName
+$password = $adminPassSecureString
+$credential = [System.Management.Automation.PSCredential]::new($username, $password)
+Connect-AzAccount -Environment AzureSampleEnvironment -Credential $credential -Tenant $azureTenantId
+
+# Undocumented 16 character limit for application names for creating service principals.
+$uniqueId = ($deploymentId -replace "-", "").Substring(0,6)
+$azureAppSpName = "sampappsp${uniqueId}"
+$azureCertSpName = "sampcertsp${uniqueId}"
+$certName = "azureSampleCert"
+$certPfxDir = [io.path]::combine($env:HOMEPATH, "cert")
+$certPfxPath = [io.path]::combine($certPfxDir, "azSampleCert.pfx")
+
+if (!(Test-Path $certPfxDir))
+{
+    New-Item -ItemType "directory" -Path $certPfxDir -Force
+}
+
+if (!$ExistingServicePrincipalDetails)
+{
+    if ($adfs)
+    {
+        $azureAppSpInfo = New-AzureAppSp -ApplicationName $azureAppSpName -AzureAdminCredential $credential
+        $azureCertSpInfo = New-AzureCertSp -ApplicationName $azureCertSpName `
+            -CertificateName $certName `
+            -AzureAdminCredential $credential `
+            -CertPfxOutputPath $certPfxPath `
+            -CertPfxPassword $SecureCertPfxPassword
+    }
+    else 
+    {
+        $azureAppSpInfo = New-AzureAppSp -ServicePrincipalName $azureAppSpName
+        $azureCertSpInfo = New-AzureCertSp -ServicePrincipalName $azureCertSpName `
+            -CertificateName $certName `
+            -CertPfxOutputPath $certPfxPath `
+            -CertPfxPassword $SecureCertPfxPassword
+    }
+}
+else
+{
+    . $ExistingServicePrincipalDetails
```

Comment:
if existing service principle details passed, this script won't work right?

Reply:
Why won't it work? It basically uses the same hash table.